clang/test/Driver/sycl-intelfpga-static-lib-win.cpp

```diff
@@ -0,0 +1,33 @@
+///
+/// tests specific to -fintelfpga -fsycl w/ static libs
+///
+// REQUIRES: clang-driver
+// REQUIRES: system-windows
+
+// make dummy archive
+// Build a fat static lib that will be used for all tests
+// RUN: echo "void foo(void) {}" > %t1.cpp
+// RUN: %clang_cl --target=x86_64-pc-windows-msvc -fsycl -fintelfpga %t1.cpp -c -o %t1_bundle.obj
+// RUN: lib -out:%t.lib %t1_bundle.obj
+
+/// Check phases with static lib
+// RUN: %clang_cl --target=x86_64-pc-windows-msvc -fsycl -fintelfpga %t.lib -ccc-print-phases 2>&1 \
+// RUN:  | FileCheck -check-prefix=CHECK_PHASES %s
+// CHECK_PHASES: 0: input, "[[INPUT:.+\.lib]]", object, (host-sycl)
+// CHECK_PHASES: 1: linker, {0}, image, (host-sycl)
+// CHECK_PHASES: 2: input, "[[INPUT]]", archive
+// CHECK_PHASES: 3: clang-offload-unbundler, {2}, archive
+// CHECK_PHASES: 4: linker, {3}, ir, (device-sycl)
+// CHECK_PHASES: 5: sycl-post-link, {4}, ir, (device-sycl)
+// CHECK_PHASES: 6: llvm-spirv, {5}, spirv, (device-sycl)
+// CHECK_PHASES: 7: input, "[[INPUT]]", archive
+// CHECK_PHASES: 8: clang-offload-unbundler, {7}, fpga_dependencies_list
+// CHECK_PHASES: 9: backend-compiler, {6, 8}, fpga_aocx, (device-sycl)
+// CHECK_PHASES: 10: clang-offload-wrapper, {9}, object, (device-sycl)
+// CHECK_PHASES: 11: offload, "host-sycl (x86_64-pc-windows-msvc)" {1}, "device-sycl (spir64_fpga-unknown-unknown-sycldevice)" {10}, image
+
+/// Check for unbundle and use of deps in static lib
+// RUN: %clang_cl --target=x86_64-pc-windows-msvc -fsycl -fintelfpga %t.lib -### 2>&1 \
+// RUN:  | FileCheck -check-prefix=CHECK_UNBUNDLE %s
+// CHECK_UNBUNDLE: clang-offload-bundler" "-type=aoo" "-targets=sycl-fpga_dep" "-inputs={{.*}}" "-outputs=[[DEPFILES:.+\.txt]]" "-unbundle"
```

Comment:
Just wondering, why is it that separate unbundler calls are being made using the same archive instead of a single one as a comma? I can't quite trace this back in the code...

Reply:
The unbundling of the dep files is separate from the unbundle call after the partial-link.  The behavior is similar to what we do for grabbing the dep files for objects.

Reply:
Might've formulated this incorrectly - didn't mean this thread to be strictly focused on the current patch :) More of a "why do we have to have separate unbundler calls at all if we're able to get multiple outputs from a single one". But I have a vague idea that the `Action` construction could be problematic, given the nature of the hack that allows us to have multiple unbundler outputs at all.

Reply:
Right.  The additional dependency target doesn't fit within the regular unbundler behavior expectations (full toolchains for each target), so we add the dep file extraction as a singular input to the offline compile.  There may be a way to accomplish this, but would probably require a bit more investigation/rework.